replication.go

```diff
@@ -322,78 +302,101 @@ func NewContentManager(db *gorm.DB, api api.Gateway, fc *filclient.FilClient, tb
 		hostname:                   cfg.Hostname,
 		inflightCids:               make(map[cid.Cid]uint),
 		FailDealOnTransferFailure:  cfg.Deal.FailOnTransferFailure,
-		isDealMakingDisabled:       cfg.Deal.Disable,
+		isDealMakingDisabled:       cfg.Deal.Disabled,
 		contentAddingDisabled:      cfg.Content.DisableGlobalAdding,
 		localContentAddingDisabled: cfg.Content.DisableLocalAdding,
-		VerifiedDeal:               cfg.Deal.Verified,
 		Replication:                cfg.Replication,
 		tracer:                     otel.Tracer("replicator"),
 		DisableFilecoinStorage:     cfg.DisableFilecoinStorage,
 		IncomingRPCMessages:        make(chan *drpc.Message),
 	}
 	qm := newQueueManager(func(c uint) {
-		cm.ToCheck <- c
+		cm.toCheck(c)
 	})
 
 	cm.queueMgr = qm
 	return cm, nil
 }
 
-func (cm *ContentManager) ContentWatcher() {
-	if err := cm.reBuildStagingZones(); err != nil {
-		log.Fatalf("failed to rebuild staging zones: %s", err)
-	}
-
-	// do not process content for storage if DisableFilecoinStorage is enabled
-	if cm.DisableFilecoinStorage {
-		return
+func (cm *ContentManager) toCheck(contID uint) {
+	// if DisableFilecoinStorage is not enabled, queue content for deal making
+	if !cm.DisableFilecoinStorage {
+		cm.ToCheck <- contID
 	}
+}
 
-	if err := cm.startup(); err != nil {
-		log.Errorf("failed to recheck existing content: %s", err)
+func (cm *ContentManager) Run(ctx context.Context) {
+	// if content adding is enabled, refresh pin queue for local contents
+	if !cm.localContentAddingDisabled {
+		go func() {
+			if err := cm.refreshPinQueue(ctx, util.ContentLocationLocal); err != nil {
+				log.Errorf("failed to refresh pin queue: %s", err)
+			}
+		}()
 	}
 
-	timer := time.NewTimer(time.Minute * 5)
+	// if staging buckets is enabled, rebuild the buckets, and run the bucket aggregate worker
+	if cm.cfg.StagingBucket.Enabled {
+		// rebuild the staging buckets
+		if err := cm.reBuildStagingBuckets(); err != nil {
+			log.Fatalf("failed to rebuild staging buckets: %s", err)
+		}
 
-	for {
-		select {
-		case c := <-cm.ToCheck:
-			var content util.Content
-			if err := cm.DB.First(&content, "id = ?", c).Error; err != nil {
-				log.Errorf("finding content %d in database: %s", c, err)
-				continue
+		// run the bucket aggregator worker
+		go func() {
+			timer := time.NewTimer(time.Minute * 5)
+			for {
+				select {
+				case <-timer.C:
+					log.Debugw("content check queue", "length", len(cm.queueMgr.queue.elems), "nextEvent", cm.queueMgr.nextEvent)
+
+					buckets := cm.popReadyStagingZone()
+					for _, b := range buckets {
+						if err := cm.aggregateContent(context.TODO(), b); err != nil {
```

Comment:
We use `context.Background` more often on the code and [`context.Background()` and `context.TODO()` return the same context object](https://www.mongodb.com/community/forums/t/go-why-context-todo-and-not-context-background/5009/2) so let's just stick with Background
```suggestion
						if err := cm.aggregateContent(context.Background(), b); err != nil {
```